builds/e2e/templates/e2e-run.yaml

```diff
@@ -0,0 +1,54 @@
+steps:
```

Comment:
is it better to use template parameter to pass values into template, such as $(arch)?

Reply:
Yeah there's definitely an argument to be made for moving `os`, `arch`, and `artifactDir` out of the top-level jobs, and into the templates. But it would make the YAML a lot more verbose and cluttered--you'd have to specify the arguments to the templates at all six template calls, vs just three jobs. Since the two options are functionally equivalent, I went with the simpler, cleaner one.